### PR TITLE
Change/picking strategy

### PR DESCRIPTION
**Pull request by karlthepagan**
Supersedes #683

### CODE REVIEW

**Comment by karlthepagan** [comment]
rebased onto the rebased #683

File: creep.behaviour.remoteMiner.js [hunk 1]
```diff
@@ -38,3 +38,20 @@ mod.approach = function(creep){
     }
     return range;
 };
+mod.strategies = {
+    defaultStrategy: {
+        name: `default-${mod.name}`,
+        // TODO don't move more than a few tiles to withdraw or pick
+        canWithdrawEnergy: function (creep) {
```
Comment (chobobobo) [inline creep.behaviour.remoteMiner.js]: Is this going to make remoteMiners wander off currently?

File: creep.behaviour.remoteHauler.js [hunk 2]
```diff
@@ -96,3 +96,43 @@ mod.gotoTargetRoom = function(creep){
...
+        name: `default-${mod.name}`,
+        canWithdrawEnergy: function (creep) {
+            const min = Math.min(creep.carryCapacity - creep.sum, 250);
+
+            return function (amount) {
+                return amount >= min;
+            };
+        }
+    },
+    uncharging: {
```
Comment (chobobobo) [inline creep.behaviour.remoteHauler.js]: Is this identical to hauler, should it perhaps just be set to the hauler's strategy instead?

**Comment by chobobobo** [comment]
@karlthepagan is this PR on hold?  If so, I will proceed with merging #1052 as it is required for SKMining haulers to pick up minerals

**Comment by karlthepagan** [comment]
There's too many existing issues to test in 0.7.0 but I think this can fit into 0.7.1 if the change set is small.

The major issue is getting more new rooms and constructions observed.

**Pull request closed without merging**